src/components/todo/all-list-todo.tsx

```diff
@@ -116,24 +50,47 @@ export default function AllListTodo({
       if (filter === "todo") return todo.isDone === false;
       return true;
     })
-    .slice(0, maxItems);
+    .sort((a, b) => {
+      const updatedDiff =
+        new Date(b.updatedAt).getTime() - new Date(a.updatedAt).getTime();
+      if (updatedDiff !== 0) return updatedDiff;
+      return new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime();
+    });
 
   return (
-    <div className="mx-auto min-h-[2080] w-full max-w-2xl rounded-xl rounded-b-none border-slate-300 bg-white p-20 text-sm text-slate-800">
-      <Filter filter={filter} setFilter={setFilter} />
-      <ul className="flex flex-col gap-20">
-        {todos.map((todo, index) => (
-          <ListTodo
-            key={todo.id}
-            index={index}
-            todo={todo}
-            toggleStatus={toggleStatus}
-            showDropdownOnHover // 드롭다운 호버 여부
-            showGoal // 목표 보여주기 여부
-          />
-        ))}
-      </ul>
-      <div ref={ref} className="h-1" />
-    </div>
+    <>
+      <div className="mb-16 mt-24 flex items-center justify-between">
+        <h2 className="text-18 font-semibold text-slate-600">{`모든 할 일 (${todos.length})`}</h2>
+        <button
+          onClick={() => setIsModalOpen(true)}
+          className="flex items-center gap-3 font-semibold text-dark-blue-600"
+        >
+          <PlusIcon />
+          <span className="text-14/3">할일 추가</span>
+        </button>
+      </div>
+      <div className="min-h-[2080px] w-full rounded-xl rounded-b-none border-slate-300 bg-white p-20 text-sm text-slate-800">
+        <Filter filter={filter} setFilter={setFilter} />
+        <ul className="flex flex-col gap-20">
```

Comment:
아이템 사이 간격이 너무 넓어서 디자인대로 8px 맞춰주시면 좋을 것 같아요!

```suggestion
        <ul className="flex flex-col gap-8">
```